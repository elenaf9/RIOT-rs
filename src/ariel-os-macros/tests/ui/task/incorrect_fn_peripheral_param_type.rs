#![no_main]
#![feature(impl_trait_in_assoc_type)]
#![feature(used_with_arg)]

// FAIL: the function is expected to take a type having a `take_peripherals()` method as first
// parameter
#[ariel_os::task(autostart, peripherals)]
async fn main(_foo: Bar) {}

struct Bar;
