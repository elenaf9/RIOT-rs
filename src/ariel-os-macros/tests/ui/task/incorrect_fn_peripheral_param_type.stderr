error[E0599]: no method named `take_peripherals` found for mutable reference `&mut OptionalPeripherals` in the current scope
 --> tests/ui/task/incorrect_fn_peripheral_param_type.rs:7:1
  |
7 | #[ariel_os::task(autostart, peripherals)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ method not found in `&mut OptionalPeripherals`
  |
  = note: this error originates in the attribute macro `ariel_os::task` (in Nightly builds, run with -Z macro-backtrace for more info)
